### PR TITLE
controlnet  canny2img

### PR DESCRIPTION
**Pull request by congw729**
Now it can run on Ascend/Pynative/MS2.0 with appropriate results.

Max L1 error:
Control model 3e-4.
Final latent code: 1e-2~1e-1.

### CODE REVIEW

File: examples/stable_diffusion_v2/canny2image.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    strength = 1 # 1~2
+    guess_mode = False
+    low_threshold =  100 #20 # 1~255
+    high_threshold = 200 #100 # 1~255
+    ddim_steps = 20 # 1~100
+    scale = 9.0 # 0.1~30
+    eta = 0.0
+    # a_prompt = 'best quality, extremely detailed' 
+    a_prompt = 'best quality'
+    n_prompt = 'longbody, lowres, bad anatomy, bad hands, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality'
```
Comment (DongyaoZhu) [inline examples/stable_diffusion_v2/canny2image.py:78]: plz move to args

File: examples/stable_diffusion_v2/canny2image.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    c_crossattn = model.get_learned_conditioning(model.tokenize([prompt + ', ' + a_prompt] * num_samples))
+    cond = {"c_concat": [control], "c_crossattn": [c_crossattn]}
+
+    un_cond_c_crossattn = model.get_learned_conditioning(model.tokenize([n_prompt] * num_samples))
+    un_cond = {"c_concat": None if guess_mode else [control], "c_crossattn": [un_cond_c_crossattn]}
+    
+    shape = (4, H // 8, W // 8)
+    model.control_scales = [strength * (0.825 ** float(12 - i)) for i in range(13)] if guess_mode else ([strength] * 13)  # Magic number. IDK why. Perhaps because 0.825**12<0.01 but 0.826**12>0.01
+    
+    logger.info('Start do inference')
```
Comment (DongyaoZhu) [inline examples/stable_diffusion_v2/canny2image.py:103]: delete "do"

**Review by DongyaoZhu — CHANGES_REQUESTED** [review]
I feel like having controlnet.py with modes "canny", "depth", etc.

**Comment by congw729** [comment]
> I feel like having controlnet.py with modes "canny", "depth", etc.
Already add params "mode" to control the mode.

**Comment by wtomin** [comment]
It's better to have a readme file as the tutorial for canny2controlnet, or for the series of controlnet models.

**Comment by congw729** [comment]
> It's better to have a readme file as the tutorial for canny2controlnet, or for the series of controlnet models.

Planning to do it after the related codes are merged.

**Review by DongyaoZhu — APPROVED** [review]

**Pull request merged**